### PR TITLE
GitHub Workflow to build the wheel

### PR DESCRIPTION
**Pull request by riedgar-ms**
Adding a GitHub workflow to run `build_wheel.py`. Ideally, the 'test' portion would also run the tests, but agreement is needed on the developer setup procedures first.

### CODE REVIEW

File: .github/workflows/python-wheel-build.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      uses: actions/upload-artifact@v3
+      with:
+        name: wheel-${{ matrix.os }}-${{ matrix.python-version }}
+        path: dist/adatest-0.3.4-py3-none-any.whl
+
+  test:
+    needs: build
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
```
Comment (riedgar-ms) [inline .github/workflows/python-wheel-build.yaml:43]: If we want to really test things, we could have each Python/OS combination test each of the wheels built in the previous step :-)

**Comment by riedgar-ms** [comment]
Ping @harsha-nori

**Comment by riedgar-ms** [comment]
Ping @Harsha-Nori @slundberg @Nking92

File: .github/workflows/python-wheel-build.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      run: |
+        pip install --upgrade pip
+        pip install --upgrade setuptools wheel
+    - name: Run build_wheel script
+      run: python scripts/build_wheel.py
+    - name: Save wheel artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: wheel-${{ matrix.os }}-${{ matrix.python-version }}
+        path: dist/adatest-0.3.4-py3-none-any.whl
```
Comment (slundberg) [inline .github/workflows/python-wheel-build.yaml:37]: Does this mean we need to update this file every time the version of adatest changes? If so we should probably load this from adatest.__version__ instead

File: .github/workflows/python-wheel-build.yaml (hunk 1 above)
Comment (slundberg) [reply .github/workflows/python-wheel-build.yaml:43]: probably not needed for now :)

File: .github/workflows/python-wheel-build.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    - name: Remove AdaTest source directory
+      uses: JesseTG/rm@v1.0.3
+      with:
+        path: adatest
+    - name: Download wheel artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: wheel-${{ matrix.os }}-${{ matrix.python-version }}
+    - name: Install wheel from file
+      run: pip install adatest-0.3.4-py3-none-any.whl
```
Comment (slundberg) [inline .github/workflows/python-wheel-build.yaml:62]: some note as above, we should probably load the right version from the package so it is not hard coded here.

**Review by slundberg — CHANGES_REQUESTED** [review]
Looks good to me, the only change I think we should have is making sure the version of adatest is not hard coded here. Thanks!

File: .github/workflows/python-wheel-build.yaml (hunk 2 above)
Comment (riedgar-ms) [reply .github/workflows/python-wheel-build.yaml:37]: Thanks for pointing this out. You're right that we wouldn't want to update this workflow every time; that would defeat the point of it.

File: .github/workflows/python-wheel-build.yaml (hunk 3 above)
Comment (riedgar-ms) [reply .github/workflows/python-wheel-build.yaml]: Fixed

**Comment by Nking92** [comment]
@riedgar-ms to fix your build, remove this line from client/package.json:

    "react-sortable-tree": "^2.8.0",

**Comment by riedgar-ms** [comment]
Thanks @Nking92 that seems to have fixed it.

File: development/scripts/install_from_wheel.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    wheel_list = list(target_dir.glob("adatest*.whl"))
+    assert len(wheel_list) == 1, f"Bad wheel_list: {wheel_list}"
+    wheel_path = wheel_list[0].resolve()
+    msg = f"Path to wheel: {wheel_path}"
+    _logger.info(msg)
+
+    _logger.info("Installing wheel")
+    # Use this approach so that extras can be added
+    adatest_spec = f"adatest[dev] @ {wheel_path.as_uri()}"
+    subprocess.run(["pip", "install", f"{adatest_spec}"], check=True)
```
Comment (riedgar-ms) [inline development/scripts/install_from_wheel.py:40]: This script is mainly about building this `pip install` command. I've found it rather tricky to do just in the shell, given how wildcarding etc. can change.

**Review by slundberg — APPROVED** [review]

**Pull request merged**